### PR TITLE
Recruit form

### LINKED ISSUE #201: New Recruit Form does not check against current recuirt nicknames
# Describe the bug
The get in use user data method does not check any in use recruit data as recruits are listed in inactive reserves.

### LINKED ISSUE #202: New recuirt form will submit with blank usernames and/or blank IDs
# Describe the bug
You can submit the form without any input. Needs validation on the data values.

### PR DESCRIPTION
**Pull request by power-5000**
# Describe the PR
Updated data validation for New Recruit Form in AccountCreation.razor on Submit. Updated GetInUseUser

# Closes
Closes #202
Closes #201 

# Changes Made
Describe your changes in detail here:
1. Submit form now validates against ValidNewID and ValidNewNickname
2. ValidNewID validates against a 5 number birthnumber
3. GetInUseUserDataAsync includes Inactive Reserves, still excludes Archived.

# Notes
Any other notes go here.

### CODE REVIEW

File: FiveOhFirstDataCore/Pages/Personnel/Recruiter/AccountCreation.razor [hunk 1]
```diff
@@ -156,16 +156,23 @@
     protected void IdChanged(ChangeEventArgs e)
     {
         string str = (string?)e.Value ?? "";
-        if(int.TryParse(str, out int val))
+        if (System.Text.RegularExpressions.Regex.IsMatch(str, @"\d\d\d\d\d"))
```
Comment (Soyvolon) [inline FiveOhFirstDataCore/Pages/Personnel/Recruiter/AccountCreation.razor:159]: You don't need a regex statement for checking if a string is blank. Just use `string.IsNullOrWhiteSpace(str)`.

File: FiveOhFirstDataCore/Pages/Personnel/Recruiter/AccountCreation.razor [hunk 2]
```diff
@@ -211,17 +218,31 @@
...
+                        TrooperData = new();
+                    }
+                    else
+                    {
+                        Errors.AddRange(errors);
+                    }  
+                }
+                else
+                {
+                    _alert.PostAlert(this, $"Trooper name is invalid");
```
Comment (Soyvolon) [inline FiveOhFirstDataCore/Pages/Personnel/Recruiter/AccountCreation.razor:240]: The overload for `_alert.PostAlert(this, "a string");` is for success messages. The overload for `_alert.PostAlert(this, new List<string> { "a string" });` is for error messages. There is a third overload that takes a alert config class for further customization.

File: FiveOhFirstDataCore.Core/Services/RosterService.cs [hunk 3]
```diff
@@ -87,7 +87,7 @@ public async Task<(HashSet<int>, HashSet<string>)> GetInUseUserDataAsync()
             {
                 ids.Add(x.Id);
 
-                if (x.Slot < Data.Slot.InactiveReserve)
```
Comment (Soyvolon) [inline FiveOhFirstDataCore.Core/Services/RosterService.cs:90]: Inactive reserves are not taking into account when getting nickname data. A new trooper can pick a nickname that someone in inactive reserves currently has.

**Review by Soyvolon — CHANGES_REQUESTED** [review]

File: FiveOhFirstDataCore.Core/Services/RosterService.cs (hunk 3 above)
Comment (power-5000) [reply FiveOhFirstDataCore.Core/Services/RosterService.cs:90]: Confirmed with Recuiter InactiveReserves are protected

File: FiveOhFirstDataCore/Pages/Personnel/Recruiter/AccountCreation.razor (hunk 2 above)
Comment (power-5000) [reply FiveOhFirstDataCore/Pages/Personnel/Recruiter/AccountCreation.razor:240]: Corrected

File: FiveOhFirstDataCore/Pages/Personnel/Recruiter/AccountCreation.razor (hunk 1 above)
Comment (power-5000) [reply FiveOhFirstDataCore/Pages/Personnel/Recruiter/AccountCreation.razor:159]: Corrected

**Review by Soyvolon — APPROVED** [review]

**Pull request merged**